### PR TITLE
Fix dependency conflict after recent Maven resolver version bumps

### LINKED ISSUE #115: Dependency conflicts in Pomegranate consumers
I’m investigating technomancy/leiningen#2567. When I use current Pomegranate master (93b1d5fcfb1401cf1f84693a0e41c7ccdcc46f24) in Leiningen I get a large list of dependency conflicts.

I noticed a couple of ‘bonus’ version bumps in the recent commit a788dec827511063ebdb9b8b44174841519d043b that look rather arbitrary – is this really necessary? It seems to break Pomegranate consumers because the different Maven artefacts are using conflicting versions.

We can tweak the Pomegranate dependencies using exclusions of course, but for now I suggest undoing the version bumps not related to Guava in a788dec827511063ebdb9b8b44174841519d043b.

Output on current Leiningen master when it uses current Pomegranate master:

    Possibly confusing dependencies found:
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven/maven-resolver-provider "3.6.0"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.1"]
     overrides
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-connector-basic "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-util "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-connector-basic "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-wagon "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-util "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-wagon "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-http "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-util "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-http "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-file "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-connector-basic "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-wagon "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-http "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-file "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]

    Consider using these exclusions:
    [leiningen-core "2.9.1" :exclusions [org.apache.maven.resolver/maven-resolver-api]]

    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven/maven-resolver-provider "3.6.0"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.1"]
     overrides
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-connector-basic "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-wagon "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-http "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-file "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.3"]

    Consider using these exclusions:
    [leiningen-core "2.9.1" :exclusions [org.apache.maven.resolver/maven-resolver-spi]]

    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven/maven-resolver-provider "3.6.0"] -> [org.apache.maven.resolver/maven-resolver-util "1.3.1"]
     overrides
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-connector-basic "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-util "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-wagon "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-util "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-http "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-util "1.3.3"]

    Consider using these exclusions:
    [leiningen-core "2.9.1" :exclusions [org.apache.maven.resolver/maven-resolver-util]]

    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-http "1.3.3"] -> [org.apache.httpcomponents/httpcore "4.4.10"]
     overrides
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.wagon/wagon-http "3.3.2"] -> [org.apache.maven.wagon/wagon-http-shared "3.3.2"] -> [org.apache.httpcomponents/httpclient "4.5.7" :exclusions [commons-logging]] -> [org.apache.httpcomponents/httpcore "4.4.11"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.wagon/wagon-http "3.3.2"] -> [org.apache.httpcomponents/httpclient "4.5.7" :exclusions [commons-logging]] -> [org.apache.httpcomponents/httpcore "4.4.11"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.wagon/wagon-http "3.3.2"] -> [org.apache.maven.wagon/wagon-http-shared "3.3.2"] -> [org.apache.httpcomponents/httpcore "4.4.11"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.httpcomponents/httpclient "4.5.7"] -> [org.apache.httpcomponents/httpcore "4.4.11"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.wagon/wagon-http "3.3.2"] -> [org.apache.httpcomponents/httpcore "4.4.11"]

    Consider using these exclusions:
    [leiningen-core "2.9.1" :exclusions [org.apache.httpcomponents/httpcore]]

    Aborting due to :pedantic? :abort
    Possibly confusing dependencies found:
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven/maven-resolver-provider "3.6.0"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.1"]
     overrides
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-connector-basic "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-util "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-connector-basic "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-wagon "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-util "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-wagon "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-http "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-util "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-http "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-file "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-connector-basic "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-wagon "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-http "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-file "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-api "1.3.3"]

    Consider using these exclusions:
    [leiningen-core "2.9.1" :exclusions [org.apache.maven.resolver/maven-resolver-api]]

    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven/maven-resolver-provider "3.6.0"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.1"]
     overrides
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-connector-basic "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-wagon "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-http "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-file "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-spi "1.3.3"]

    Consider using these exclusions:
    [leiningen-core "2.9.1" :exclusions [org.apache.maven.resolver/maven-resolver-spi]]

    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven/maven-resolver-provider "3.6.0"] -> [org.apache.maven.resolver/maven-resolver-util "1.3.1"]
     overrides
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-connector-basic "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-util "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-wagon "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-util "1.3.3"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-http "1.3.3"] -> [org.apache.maven.resolver/maven-resolver-util "1.3.3"]

    Consider using these exclusions:
    [leiningen-core "2.9.1" :exclusions [org.apache.maven.resolver/maven-resolver-util]]

    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.resolver/maven-resolver-transport-http "1.3.3"] -> [org.apache.httpcomponents/httpcore "4.4.10"]
     overrides
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.wagon/wagon-http "3.3.2"] -> [org.apache.maven.wagon/wagon-http-shared "3.3.2"] -> [org.apache.httpcomponents/httpclient "4.5.7" :exclusions [commons-logging]] -> [org.apache.httpcomponents/httpcore "4.4.11"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.wagon/wagon-http "3.3.2"] -> [org.apache.httpcomponents/httpclient "4.5.7" :exclusions [commons-logging]] -> [org.apache.httpcomponents/httpcore "4.4.11"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.wagon/wagon-http "3.3.2"] -> [org.apache.maven.wagon/wagon-http-shared "3.3.2"] -> [org.apache.httpcomponents/httpcore "4.4.11"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.httpcomponents/httpclient "4.5.7"] -> [org.apache.httpcomponents/httpcore "4.4.11"]
     and
    [leiningen-core "2.9.1"] -> [com.cemerick/pomegranate "1.1.1-SNAPSHOT" :exclusions [org.slf4j/jcl-over-slf4j]] -> [org.apache.maven.wagon/wagon-http "3.3.2"] -> [org.apache.httpcomponents/httpcore "4.4.11"]

    Consider using these exclusions:
    [leiningen-core "2.9.1" :exclusions [org.apache.httpcomponents/httpcore]]

    Aborting due to :pedantic? :abort

### PR DESCRIPTION
**Pull request by glts**
Version 3.6.0 of Maven Resolver Provider, which was recently introduced (a788dec827511063ebdb9b8b44174841519d043b), contains transitive dependencies that conflict with each other. This breaks Pomegranate consumers like Leiningen. The conflicts are resolved in version 3.6.1 of this dependency.

The proposed change bumps Maven Resolver Provider to 3.6.1. It also adds four ‘used but undeclared’ Maven resolver dependencies that are directly used in Pomegranate, and so should be declared as dependencies. Finally, it pins the transitive dependency `org.apache.httpcomponents/httpcore`, again to prevent a version conflict.

I have verified that this fixes the breakage in the Leiningen build, so this fixes #115.

### CODE REVIEW

**Comment by glts** [comment]
@cemerick Could we merge this?

This simply fixes downstream breakage introduced by the most recently
merged pull request. We will anyway need this for the next release of
Leiningen.

**Comment by cemerick** [comment]
I'm really not in a position to usefully act on this (as might be said for said "recent" merged PRs as well, if they were breaking).

I think it's clear that my continuing to be an absent maintainer here is not helping anyone, probably hasn't for a very long time, and that it's very unlikely for this to change.

I strongly recommend that those most in need of an actively-maintained pomegranate implementation should fork this repo and continue as they see fit.

Sorry, all.

**Comment by glts** [comment]
I’m still interested in getting this merged. We need this in Leiningen.

**Comment by glts** [comment]
Force-pushed to propagate changes to deps.edn. I feel like the import of this somehow got lost in all the prose above, so let me try again in a bold font.

**Pomegranate master is broken, it cannot be used in Leiningen.**

Apologies if I'm being too insistent, but this is a 100% benign change, we'll need it in the future. It is also gentle, conservative version bumping, see also the commit message for explanation. Thank you.

**Comment by glts** [comment]
Marvellous, thank you for the merge.

**Comment by slipset** [comment]
@glts would you mind sending me an email at erik at assum dot net ?
